riot.d.ts

```diff
@@ -204,36 +204,100 @@ export declare function pure<
 export declare const version: string
 
 // typescript specific methods
-export declare function withTypes<
-  Component extends RiotComponent,
-  ComponentFactory = RiotComponentFactoryFunction<
-    AutobindObjectMethods<RiotComponentWithoutInternals<Component>, Component>
-  >,
->(fn: ComponentFactory): () => Component
-export declare function withTypes<
-  Component extends RiotComponent,
-  ComponentFactory = RiotComponentFactoryFunction<
-    AutobindObjectMethods<
-      RiotComponentWithoutInternalsAndInitialState<Component>,
-      Component
-    >
-  >,
->(fn: ComponentFactory): () => Component
-export declare function withTypes<
-  Component extends RiotComponent,
-  ComponentObjectWithInitialState = RiotComponentWithoutInternals<Component>,
->(
-  component: AutobindObjectMethods<ComponentObjectWithInitialState, Component>,
-): Component
-export declare function withTypes<
-  Component extends RiotComponent,
-  ComponentObjectWithoutInitialState = RiotComponentWithoutInternalsAndInitialState<Component>,
->(
-  component: AutobindObjectMethods<
-    ComponentObjectWithoutInitialState,
-    Component
-  >,
-): Component
+export type NeverOfKeys<T extends PropertyKey> = { [K in T]?: never }
```

Comment:
This code is extremely convoluted and hard to understand. I don't get if now that's needed for https://github.com/riot/riot/issues/3046 or to implement also the `composeTypes` method.

I'd recommend to add more tests for https://github.com/riot/riot/issues/3046 and start patching the types. I will be glad to help you finding the easiest solution for it